### ISSUE TITLE
Use treemacs outside of project directory

### ISSUE THREAD

**Issue by jcs090218** [Alexander-Miller/treemacs#773]
I tried to `M-x treemacs` but I am not able to see my current file's directory. Is this the default behaviour?

I was expecting the same as this video https://www.youtube.com/watch?v=ACg6RnVdw8k&feature=emb_logo&ab_channel=MikeZamansky. Once I called `treemacs` should just show the file directory from the variable `default-directory`.

Thanks!

**Comment by Alexander-Miller** [comment]
When you start treemacs without any previous layout it will first *ask* you for the location of your first project. You should see a dialog asking you for your "Project Root". The `default-directory` is just the preselected value, but it's not automatically accepted.

If you accidentally closed this dialog then its possible that the treemacs buffer is left empty. That is not a problem, you can still add projects with commands like `treemacs-add-and-display-current-project` or `treemacs-add-project-to-workspace`. (Don't for get to check the helpful hydra at `?` for keybinds)

You can also edit your workspace layout as an org-mode file: https://github.com/Alexander-Miller/treemacs#conveniently-editing-your-projects-and-workspaces

**Comment by jcs090218** [comment]
Can you view treeview without the project existence?

**Comment by Alexander-Miller** [comment]
Yes. Treemacs is a *file* explorer. Anything involving projects is just a convenience.

**Comment by andreyorst** [comment]
I've had discussion with some Emacs users who wanted to try out Treemacs, but were turned off by it's project notation and implementation of `follow-mode`.

The main complaint was that `follow-mode` doesn't follow files that are not present in current project, and adding tons of small projects for every file they're working on is a tedious task. I've suggested that it is possible to use their home directory as a Treemacs project, and then Treemacs will follow each file reachable from home directory, however this was not welcomed as solution, as it will clutter Treemacs buffer with way too many files.

Basically, after distilling their desired outcome, we've summarized it down to: "always show me the default-directory contents", basically meaning that Treemacs, shoud simply display contents of buffer's default directory, and when swithcing buffers the contents of treemacs should change. However, the ability to navigate filetree should be preserved, so if for some buffer, user went up or down the tree, after swithcing to different buffer and back into previous one, Treemacs should remember where user was in the tree, relatively to the default directory of that buffer.

So for example, buffer `Buffer_A` has default directory `dir_a`, and buffer `Buffer_B` has default directory `dir_b`. With `Buffer_A` as current buffer:

- user goes one level up from `dir_a` to `dir_a/../`
  - Treemacs remembers this as `Buffer_A` default-directory override
- user switches to `Buffer_B`
  - Treemacs displays `dir_b`
- user switches back to `Buffer_A`
  - Treemacs shows `dir_a/../` contents.

And if case where several buffers share default directory, if user navigates the tree, it should be remembered for all of these buffers.

This behavior doesn't require setting any project roots anywhere, so seems matching this particular discussion, but feel free to request separate issue for such feature request.
